src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -66,138 +66,152 @@ protected override SyntaxNode MakeSafeToAddNamespaces(SyntaxNode root, IEnumerab
             return null;
         }
 
-        private class Rewriter : CSharpSyntaxRewriter
+        /// <summary>
+        /// Walks the portion of the tree we're adding imports to looking to see if those imports could likely cause
+        /// conflicts with existing code.  Note: this is a best-effort basis, and the goal is to catch reasonable
+        /// conflicts effectively.  There may be cases that do slip through that we can adjust for in the future.  Those
+        /// cases should be assessed to see how reasonable/likely they are.  I.e. if it's just a hypothetical case with
+        /// no users being hit, then that's far less important than if we have a reasonable coding pattern that would be
+        /// impacted by adding an import to a normal namespace.
```

Comment:
And I'm guessing the most likely case where we screw up it'll just be an ambiguity too and the user can figure it out...less likely to have silent changes of behavior...

Reply:
right.  it's not pefect, but i think it's better in balance from what we have today.